bentoml/_internal/server/instruments.py

```diff
@@ -38,6 +38,10 @@ def _setup(
     ):
         self.metrics_client = metrics_client
         service_name = self.bento_service.name
+        # a valid tag name may includes invalid characters, so we need to escape them
+        # ref: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+        service_name = service_name.replace("-", ":").replace(".", "::")
```

Comment:
instead of ":", could we replace them with just underscore? "_"

it seems very uncommon for metric names to contain ":" https://github.com/prometheus/docs/blob/main/content/docs/practices/naming.md

Reply:
If so, there would be a collision between service "service-1" and "service_1", since "_" is valid for bentoml tag. But ":" is not. Thus it is the best replacement.